Tweetview rework

Morpog opened the pull request:
As previewed in #111
Also fixes Favorite star icon not showing up on tweetpage.

Morpog committed:
Tweetview rework

As previewed in #111
Also fixes Favorite star icon not showing up on tweetpage.

veskuh commented on qml/tweetian-harmattan/Delegate/DMThreadDelegate.qml:
```diff
@@ -57,13 +57,19 @@ AbstractDelegate {
     }
 
-    Text {
+    Item {
```
Reason for this item is just to get paddingSmall under the text. Couldn't it be defined in the tweetTime item itself?

Morpog replied:
Well I tried to add top and bottom margins there, but they got ignored. Guess it's because of that Column in AbstractDelegate. What should we do here?

veskuh replied:
Ok, this is a minor thing and can be optimized later if needed.

veskuh commented on qml/tweetian-harmattan/Delegate/DirectMsgDelegate.qml:
```diff
@@ -56,13 +56,21 @@ AbstractDelegate {
     }
 
-    Text {
+    Item {
```
Again extra item.

Morpog replied:
Same here as above.

veskuh commented on qml/tweetian-harmattan/Delegate/TweetDelegate.qml:
```diff
@@ -71,15 +69,15 @@ AbstractDelegate {
         Loader {
```
This is not related to your change, but at some point would be nice to simplify these a bit and get rid of some of these loaders. This would be a good candidate to get rid of since loader only controls if image is visible or not. Simpler just to use visible property of image and probably even performs better.

Morpog replied:
OK, my prototype was without loaders, but I thought those were by purpose and added them all again :D

veskuh commented on qml/tweetian-harmattan/TweetPage.qml:
```diff
@@ -187,19 +187,42 @@ Page {
+                                text: tweet.name
+                            }
+
+                            Loader {
```
At least lets not add new loaders.

This would just be:

```
Image {
      anchors { right: parent.right; rightMargin: constant.paddingMedium; verticalCenter: userNameText.verticalCenter 
      visible: favouritedTweet
      height: constant.graphicSizeSmall
      width: height
      source: "image://theme/icon-s-favorite"
}
```

Morpog committed:
Fix loader comment

Not sure if this is less complex, as visible property didn't de-allocate
space it used. So I had to force that with some further checks in
TweetDelegate. See line 109 and line 120.
